ios/FluentUI/Vnext/Avatar/AvatarGroup.swift

```diff
@@ -0,0 +1,191 @@
+    var maxDisplayedAvatars: Int { get set }
+    var overflowCount: Int { get set }
+    var style: MSFAvatarGroupStyle { get set }
+
+    func createAvatar(style: MSFAvatarStyle, size: MSFAvatarSize)
+    func getAvatarState(_ atIndex: Int) -> MSFAvatarState?
+    func deleteAvatar(atIndex: Int)
+}
+
+/// Properties that make up AvatarGroup content
+class MSFAvatarGroupStateImpl: NSObject, ObservableObject, MSFAvatarGroupState {
+    func createAvatar(style: MSFAvatarStyle, size: MSFAvatarSize) {
+        avatars.append(AvatarView(style: style, size: size))
+    }
+
+    func getAvatarState(_ atIndex: Int) -> MSFAvatarState? {
+        if atIndex >= avatars.count {
+            print("Requested index is out of bounds!")
+            return nil
+        } else {
+            return avatars[atIndex].state
+        }
+    }
+
+    func deleteAvatar(atIndex: Int) {
+        avatars.remove(at: atIndex)
+    }
+
+    @Published var avatars: [AvatarView] = []
+    @Published var maxDisplayedAvatars: Int = Int.max
+    @Published var overflowCount: Int = 0
+
+    var size: MSFAvatarSize {
+        get {
+            return tokens.size
+        }
+        set {
+            tokens.size = newValue
+        }
+    }
+
+    var style: MSFAvatarGroupStyle {
+        get {
+            return tokens.style
+        }
+        set {
+            tokens.style = newValue
+        }
+    }
+
+    var tokens: MSFAvatarGroupTokens
+
+    init(style: MSFAvatarGroupStyle,
+         size: MSFAvatarSize) {
+        self.tokens = MSFAvatarGroupTokens(style: style,
+                                           size: size)
+        super.init()
+    }
+}
+
+/// View that represents the AvatarGroup
+public struct AvatarGroup: View {
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    @ObservedObject var state: MSFAvatarGroupStateImpl
+    @ObservedObject var tokens: MSFAvatarGroupTokens
+
+    init(style: MSFAvatarGroupStyle, size: MSFAvatarSize) {
+        let state = MSFAvatarGroupStateImpl(style: style, size: size)
+        self.state = state
+        self.tokens = state.tokens
+    }
+
+    public var body: some View {
+        let size: CGFloat = tokens.size.size
+        let x: CGFloat = size + tokens.interspace - tokens.ringThickness
+        let avatars: [AvatarView] = state.avatars
+        let maxDisplayedAvatars: Int = avatars.prefix(state.maxDisplayedAvatars).count
+        let overflowCount: Int = (avatars.count > maxDisplayedAvatars ? avatars.count - maxDisplayedAvatars : 0) + state.overflowCount
+        HStack(spacing: 0) {
+            ForEach(0 ..< maxDisplayedAvatars, id: \.self) { index in
+                // If the avatar is part of Stack style and is not the last avatar in the sequence, create a cutout
+                let needsCutout = tokens.style == .stack && (overflowCount > 0 || index + 1 < maxDisplayedAvatars)
+                state.avatars[index]
```

Comment:
After the merge of #608, we can now create the avatar states in **createAvatar(style: MSFAvatarStyle, size: MSFAvatarSize)** and pass them to the avtars being created here instead of storing the Avatar Views in the state object.